### DISCUSSION TITLE
is there an easy way to disable particular protocols on a per-vnode basis?

### DISCUSSION THREAD

**Discussion by PatrickTague** [coreemu/core#611] [Q&A] [answered]
i'm an instructor using core in a networking course, and one of our current tasks is to implement aspects of IP and TCP using raw sockets.  core is exactly the right tool for this, because it keeps the students from messing around on the public cluster computers, but some of the default nodes and services are interfering a bit.  in particular, initiating a TCP handshake with a TCP SYN leads to an RST-ACK response from the stock TCP stack in the receiver.  is there a way to disable this TCP stack or to create a custom host that doesn't have a TCP stack?

**Comment by bharnden** [comment, marked as answer]
I am not familiar with anyone who has had this desire or if achievable. Anything you could do on a standard Linux host, you should be able to replicate, if disabling TCP like you desire is possible, for your use case.

You can disable all services for a given node, but that is just arbitrary software and should not really be relevant in what you desire. You can create a custom node, that has no services by default as well, to make that process easier.
